frontend/src/reports/ui/ShareDialog.tsx

```diff
@@ -1,23 +1,22 @@
-import React, { useState } from "react";
-import { MadLib } from "../../utils/MadLibs";
+import Button from "@material-ui/core/Button";
 import Dialog from "@material-ui/core/Dialog";
 import DialogContent from "@material-ui/core/DialogContent";
 import DialogContentText from "@material-ui/core/DialogContentText";
 import DialogTitle from "@material-ui/core/DialogTitle";
-import { getMadLibPhraseText } from "../../utils/MadLibs";
-import { linkToMadLib } from "../../utils/urlutils";
-import FileCopyIcon from "@material-ui/icons/FileCopy";
-import TextField from "@material-ui/core/TextField";
 import FormControl from "@material-ui/core/FormControl";
+import TextField from "@material-ui/core/TextField";
+import FileCopyIcon from "@material-ui/icons/FileCopy";
+import React, { useState } from "react";
 import { CopyToClipboard } from "react-copy-to-clipboard";
-import Button from "@material-ui/core/Button";
+import { getMadLibPhraseText, MadLib } from "../../utils/MadLibs";
 
 function ShareDialog(props: {
   madLib: MadLib;
   shareModalOpen: boolean;
   setShareModalOpen: (shareModalOpen: boolean) => void;
 }) {
   const [textCopied, setTextCopied] = useState(false);
+  const text = window.location.href;
```

Comment:
I'm not entirely sure how react re-rendering works with window.location.href.

I think it might be better to pass this in as a prop so that the component re-renders whenever the url changes. Probably in practice this doesn't matter since the share dialog isn't expected to change, but I think just good practice?

Reply:
Yeah this should never need to be rerendered but its just following previous behavior :).  TBH this dialogue shouldnt even exist but 🤷‍♂️